x-pack/platform/test/onechat_api_integration/apis/esql_tools.ts

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../api_integration/ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const kibanaServer = getService('kibanaServer');
+  const log = getService('log');
+
+  describe('ES|QL Tools API', () => {
+    const createdToolIds: string[] = [];
+
+    const mockTool = {
+      id: 'cases-tool',
+      type: 'esql',
+      description: 'A test tool',
+      tags: ['test'],
+      configuration: {
+        query: 'FROM my_cases | WHERE case_id == ?case_id',
+        params: { case_id: { type: 'keyword', description: 'Case ID' } },
+      },
+    };
+
+    before(async () => {
+      await kibanaServer.uiSettings.update({
+        'onechat:api:enabled': true,
+      });
+    });
+
+    after(async () => {
+      for (const toolId of createdToolIds) {
+        try {
+          await supertest.delete(`/api/chat/tools/${toolId}`).set('kbn-xsrf', 'kibana').expect(200);
+        } catch (error) {
+          log.warning(`Failed to delete tool ${toolId}: ${error.message}`);
+        }
+      }
+
+      await kibanaServer.uiSettings.update({
+        'onechat:api:enabled': false,
+      });
+    });
+
+    describe('POST /api/chat/tools', () => {
```

Comment:
Could you add a few more additional tests?
- try to delete a system tool like nl_search
- try to create a tool under an existing tool id
- cover the _execute API

Reply:
cool will create a ticket to follow up on tools FTR spec